src/fhirtypes/ElementDefinition.ts

```diff
@@ -1368,22 +1367,22 @@ export class ElementDefinition {
    * @param {InstanceDefinition} value - The resource to fix
    * @param {Fishable} fisher - A fishable implementation for finding definitions and metadata
    * @throws {MismatchedTypeError} when the ElementDefinition is not of type Resource
-   * @throws {FixingNonResourceError} when the value being fixed is non-Resource
    * @returns {InstanceDefinition} the input value when it can be fixed
    */
-  checkFixResource(value: InstanceDefinition, fisher: Fishable): InstanceDefinition {
-    if (this.type?.some(t => isInheritedResource(value.resourceType, t.code, fisher))) {
+  checkFixInlineInstance(value: InstanceDefinition, fisher: Fishable): InstanceDefinition {
+    const inlineInstanceType = value.resourceType ?? value._instanceMeta.sdType;
+    const lineage = this.getTypeLineage(inlineInstanceType, fisher).map(
+      metadata => metadata.sdType
+    );
+    if (this.type?.some(t => lineage.includes(t.code))) {
       return value;
     } else {
-      const resource = fisher.fishForFHIR(value.resourceType, Type.Resource);
-      if (resource) {
-        // In this case the value is actually a resource, but not one that inherits from this.type
-        const typesString = this.type?.map(t => t.code).join(', ');
-        throw new MismatchedTypeError(value.resourceType, value.id, typesString);
-      } else {
-        // In this case the value is not actually a resource
-        throw new FixingNonResourceError(value.resourceType, value.id);
-      }
+      // In this case the value is actually a resource, but not one that inherits from this.type
```

Comment:
Now it's not necessarily a _resource_, right?

Reply:
It wasn't a big deal, but I couldn't help fixing it now that I see it. So it's fixed in https://github.com/FHIR/sushi/pull/464/commits/c14eea953273941b738d1421af2192da8c56b20e, sorry to dismiss your review 😬